ksql-parser/src/test/java/io/confluent/ksql/parser/rewrite/StatementRewriterTest.java

```diff
@@ -0,0 +1,616 @@
+    StatementRewriter statementRewriter = new StatementRewriter();
+    Statement rewrittenStatement = (Statement) statementRewriter.process(statement, null);
+
+    assertThat("testSimpleQuery fails", rewrittenStatement, instanceOf(Query.class));
+
+    Query query = (Query) rewrittenStatement;
+    assertThat("testProjectFilter fails", query.getQueryBody(), instanceOf(QuerySpecification.class));
+    QuerySpecification querySpecification = (QuerySpecification)query.getQueryBody();
+
+    assertThat("testProjectFilter fails", querySpecification.getWhere().get(), instanceOf(ComparisonExpression.class));
+    ComparisonExpression comparisonExpression = (ComparisonExpression)querySpecification.getWhere().get();
+    assertThat("testProjectFilter fails", comparisonExpression.toString(), equalTo("(TEST1.COL0 > 100)"));
+    assertThat("testProjectFilter fails", querySpecification.getSelect().getSelectItems().size(), equalTo(3));
+
+  }
+
+  @Test
+  public void testBinaryExpression() {
+    String queryStr = "SELECT col0+10, col2, col3-col1 FROM test1;";
+    Statement statement = KSQL_PARSER.buildAst(queryStr, metaStore).get(0);
+    StatementRewriter statementRewriter = new StatementRewriter();
+    Statement rewrittenStatement = (Statement) statementRewriter.process(statement, null);
+
+    assertThat("testSimpleQuery fails", rewrittenStatement, instanceOf(Query.class));
+
+    Query query = (Query) rewrittenStatement;
+    assertThat("testBinaryExpression fails", query.getQueryBody(), instanceOf(QuerySpecification.class));
+    QuerySpecification querySpecification = (QuerySpecification)query.getQueryBody();
+    SingleColumn column0 = (SingleColumn)querySpecification.getSelect().getSelectItems().get(0);
+    assertThat("testBinaryExpression fails", column0.getAlias().get(), equalTo("KSQL_COL_0"));
+    assertThat("testBinaryExpression fails", column0.getExpression().toString(), equalTo("(TEST1.COL0 + 10)"));
+  }
+
+  @Test
+  public void testBooleanExpression() {
+    String queryStr = "SELECT col0 = 10, col2, col3 > col1 FROM test1;";
+    Statement statement = KSQL_PARSER.buildAst(queryStr, metaStore).get(0);
+    StatementRewriter statementRewriter = new StatementRewriter();
+    Statement rewrittenStatement = (Statement) statementRewriter.process(statement, null);
+
+    assertThat("testSimpleQuery fails", rewrittenStatement, instanceOf(Query.class));
+
+    Query query = (Query) rewrittenStatement;
+    assertThat("testProjection fails", query.getQueryBody(), instanceOf(QuerySpecification.class));
+    QuerySpecification querySpecification = (QuerySpecification)query.getQueryBody();
+    SingleColumn column0 = (SingleColumn)querySpecification.getSelect().getSelectItems().get(0);
+    assertThat("testBooleanExpression fails", column0.getAlias().get(), equalTo("KSQL_COL_0"));
+    assertThat("testBooleanExpression fails", column0.getExpression().toString(), equalTo("(TEST1.COL0 = 10)"));
```

Comment:
It would be good to validate the type of the returned Expression. Ditto below.

Reply:
I added check for the expression class but checking the types (ksql types) is not necessary here since we just rewrite the tree.